pkgs/mime/lib/src/magic_numbers.dart

```diff
@@ -0,0 +1,154 @@
+  MagicNumber(
+    'audio/x-wav',
+    hex('524946460000000057415645'),
+    mask: hex('FFFFFFFF00000000FFFFFFFF'),
+  ),
+  MagicNumber('image/gif', hex('474946383761')),
+  MagicNumber('image/gif', hex('474946383961')),
+  MagicNumber('image/jpeg', hex('FFD8')),
+  MagicNumber('image/png', hex('89504E470D0A1A0A')),
+  MagicNumber('image/tiff', hex('49492A00')),
+  MagicNumber('image/tiff', hex('4D4D002A')),
+  MagicNumber('audio/aac', hex('FFF1')),
+  MagicNumber('audio/aac', hex('FFF9')),
+  MagicNumber('audio/weba', hex('1A45DFA3')),
+  MagicNumber('audio/mpeg', hex('494433')),
+  MagicNumber('audio/mpeg', hex('FFFB')),
+  MagicNumber('audio/ogg', hex('4F7075')),
+  MagicNumber(
+    'video/3gpp',
+    hex('000000006674797033677035'),
+    mask: hex('FFFFFF00FFFFFFFFFFFFFFFF'),
+  ),
+  MagicNumber(
+    'video/mp4',
+    hex('000000006674797061766331'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+  MagicNumber(
+    'video/mp4',
+    hex('000000006674797069736F32'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+  MagicNumber(
+    'video/mp4',
+    hex('000000006674797069736F6D'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+  MagicNumber(
+    'video/mp4',
+    hex('00000000667479706D703431'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+  MagicNumber(
+    'video/mp4',
+    hex('00000000667479706D703432'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+  MagicNumber('model/gltf-binary', hex('46546C67')),
+
+  /// The WebP file format is based on the RIFF document format.
+  /// -> 4 bytes have the ASCII characters 'R' 'I' 'F' 'F'.
+  /// -> 4 bytes indicating the size of the file
+  /// -> 4 bytes have the ASCII characters 'W' 'E' 'B' 'P'.
+  /// https://developers.google.com/speed/webp/docs/riff_container
+  MagicNumber(
+    'image/webp',
+    hex('524946460000000057454250'),
+    mask: hex('FFFFFFFF00000000FFFFFFFF'),
+  ),
+
+  MagicNumber('font/woff2', hex('774f4632')),
+
+  /// High Efficiency Image File Format (ISO/IEC 23008-12).
+  /// -> 4 bytes indicating the ftyp box length.
+  /// -> 4 bytes have the ASCII characters 'f' 't' 'y' 'p'.
+  /// -> 4 bytes have the ASCII characters 'h' 'e' 'i' 'c'.
+  /// https://www.iana.org/assignments/media-types/image/heic
+  MagicNumber(
+    'image/heic',
+    hex('000000006674797068656963'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+
+  /// -> 4 bytes indicating the ftyp box length.
+  /// -> 4 bytes have the ASCII characters 'f' 't' 'y' 'p'.
+  /// -> 4 bytes have the ASCII characters 'h' 'e' 'i' 'x'.
+  MagicNumber(
+    'image/heic',
+    hex('000000006674797068656978'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+
+  /// -> 4 bytes indicating the ftyp box length.
+  /// -> 4 bytes have the ASCII characters 'f' 't' 'y' 'p'.
+  /// -> 4 bytes have the ASCII characters 'm' 'i' 'f' '1'.
+  MagicNumber(
+    'image/heif',
+    hex('00000000667479706D696631'),
+    mask: hex('00000000FFFFFFFFFFFFFFFF'),
+  ),
+];
+
+Uint8List hex(String encoded) {
```

Comment:
Consider dropping the `hex` and allowing the list to be constant.
Use String hex encodings like:
```dart
MagicNumber(
    'image/heif',
    '\x00\x00\x00\x00\x66\x74\x79\x70\x6D\x69\x66\x31',
    mask: '\x00\x00\x00\x00\xFF\xFF\xFF\xFF\xFF\xFF\xFF\xFF',
),
```
Having the list be constant means you won't have all the computation of converting every string to a `Uint8List` on first access.
Then work directly off the strings, using `codeUnitAt` instead of `List.[]`.

(I considered making the hex-parsing lazy, but you're likely to access every element if you trry to magic-number a random file.)